### ISSUE TITLE
Compression dictionary transport

### ISSUE THREAD

**Issue by captainbrosset** [web-platform-dx/web-features#1944]
### Specification

https://datatracker.ietf.org/doc/draft-ietf-httpbis-compression-dictionary/

### Description

I was doing some quick research on compression dictionary and didn't want to lose what I had found on the topic, but also didn't have time to create a new issue right away for it.

- "Intent To": https://groups.google.com/a/chromium.org/g/blink-dev/c/7YdP2YxRQn4/m/9oPxLDdjAAAJ
- Chromestatus: https://chromestatus.com/feature/5124977788977152
- Origin Trial: https://developer.chrome.com/origintrials/#/view_trial/3693514644397228033
- CrBug: https://issues.chromium.org/issues/40255884
- WebKit position: https://github.com/WebKit/standards-positions/issues/160 (support)
- Mozilla position: https://github.com/mozilla/standards-positions/issues/771 (positive)
- Bugzilla tracking bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1882979
- Blog post: https://developer.chrome.com/blog/shared-dictionary-compression

Shipping in Chrome/Edge in 130. Not implemented in other engines, but positive positions.

Compression shared dictionaries can bring huge improvements in the amount of data transmitted over the network.
Worth tracking in web-features as this is an upcoming feature.

**Comment by tunetheweb** [comment]
If it helps, this has been added to MDN BCD in https://github.com/mdn/browser-compat-data/pull/26257 and included in [v6.0.1](https://github.com/mdn/browser-compat-data/releases/tag/v6.0.1). But it's a collection of various HTTP headers and HTML element attributes, so would be nice to map them to one overall feature.

**Comment by tunetheweb** [comment]
Looks like this is what we need:

```yaml
name: Compression Dictionary Transport
description: Compression Dictionary Transport is a mechanism for using designated HTTP responses as an external dictionary for future HTTP responses for compression schemes that support using external dictionaries (e.g., Brotli and Zstandard)
spec: https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-compression-dictionary
status:
  compute_from:
    - html.elements.link.compression-dictionary
    - http.headers.accept-encoding.dcb
    - http.headers.accept-encoding.dcz
    - http.headers.content-encoding.dcb
    - http.headers.content-encoding.dcz
    - http.headers.available-dictionary
    - http.headers.dictionary-id
    - http.headers.link.compression-dictionary
    - http.headers.use-as-dictionary
```

But will need to wait for BCD v6.0.1 to be available in this repo before this can be added. And in particular upgrading to v6 of the BCD package which [has some breaking changes](https://github.com/mdn/browser-compat-data/releases/tag/v6.0.0).

**Comment by captainbrosset** [comment]
Thanks @tunetheweb. I'm bumping this project to BCD 6.0.1. When I'm done, the new keys will appear in features/draft/spec/compression-dictionary.yml which we should work into converting to a real (non-draft) feature, based on your last comment.

**Comment by captainbrosset** [comment]
@tunetheweb, please see this draft feature:

https://github.com/web-platform-dx/web-features/blob/3b777408f530616c6e6ac25ed1445aaac3e315c0/features/draft/spec/compression-dictionary.yml#L1-L14

Would you be interested in opening a PR to move it to a real feature, and incorporate the changes you mentioned before?
In case that helps: https://github.com/web-platform-dx/web-features/blob/main/docs/CONTRIBUTING.md#create-a-new-feature-from-a-draft-feature

**Comment by captainbrosset** [comment]
> BTW curious as to why it's compression-dictionary.yml and not compression-dictionary-transport.yml? Is that based on the last part of the draft-ietf-httpbis-compression-dictionary.html?

You can safely ignore the feature ID (filename). Things under `/features/draft/spec/` get generated from specs. So, while they contain useful information, they're usually not at the right level of granularity (thought this one might be), and their names are generated. If you have a better name, please go with it.
We want IDs, names, descriptions, and compat data to match with how developers perceive this feature. Not how it's spec'd.

> Anyway, will submit a PR as soon as I can now I'm unblocked on this repos side anyway.

Excellent! Thank you so much.

**Comment by tunetheweb** [comment]
#2836 raised! PTAL
